tests/build/build.go

```diff
@@ -1359,4 +1359,240 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+					}
+					return nil
+				}, timeout, constants.PipelineRunPollingInterval).Should(Succeed(), fmt.Sprintf("timed out when waiting for the PipelineRun to start for the component %s/%s", ParentComponentDef.componentName, testNamespace))
+			})
+			It(fmt.Sprintf("the PipelineRun should eventually finish successfully for component %s", ParentComponentDef.componentName), func() {
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(ParentComponentDef.component, "", f.AsKubeAdmin.TektonController, &has.RetryOptions{Always: true, Retries: 2})).To(Succeed())
+				pr, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(ParentComponentDef.component.GetName(), ParentComponentDef.component.Spec.Application, ParentComponentDef.component.GetNamespace(), "")
+				Expect(err).ShouldNot(HaveOccurred())
+				for _, result := range pr.Status.PipelineResults {
+					if result.Name == "IMAGE_DIGEST" {
+						parentFirstDigest = result.Value.StringVal
+					}
+				}
+				Expect(parentFirstDigest).ShouldNot(BeEmpty())
+			})
+
+			It("create dockerfile to update referencing build repository", func() {
+
+				component, err := f.AsKubeAdmin.HasController.GetComponent(ParentComponentDef.componentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred(), "could not get component %s in the %s namespace", ParentComponentDef.componentName, testNamespace)
+
+				annotations := component.GetAnnotations()
+				imageRepoName, err := build.GetQuayImageName(annotations)
+				Expect(err).ShouldNot(HaveOccurred())
+				err = f.AsKubeAdmin.CommonController.Github.CreateRef(ChildComponentDef.repoName, ChildComponentDef.baseBranch, ChildComponentDef.baseRevision, ChildComponentDef.pacBranchName)
+				Expect(err).ShouldNot(HaveOccurred())
+				_, err = f.AsKubeAdmin.CommonController.Github.CreateFile(ChildComponentDef.repoName, "Dockerfile.tmp", "FROM quay.io/"+gihubOrg+"/"+imageRepoName+"@"+parentFirstDigest+"\nRUN echo hello\n", ChildComponentDef.pacBranchName)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				_, err = f.AsKubeAdmin.CommonController.Github.CreatePullRequest(ChildComponentDef.repoName, "update to build repo image", "update to build repo image", ChildComponentDef.pacBranchName, ChildComponentDef.componentBranch)
+				Expect(err).ShouldNot(HaveOccurred())
+				prs, err := f.AsKubeAdmin.CommonController.Github.ListPullRequests(ChildComponentDef.repoName)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				prno := -1
+				found := false
+				for _, pr := range prs {
+					if pr.Head.GetRef() == ChildComponentDef.pacBranchName {
+						prno = pr.GetNumber()
+						found = true
+					}
+				}
+				Expect(found).To(BeTrue())
+				_, err = f.AsKubeAdmin.CommonController.Github.MergePullRequest(ChildComponentDef.repoName, prno)
+				Expect(err).ShouldNot(HaveOccurred())
+
+			})
+			It("should lead to a PaC PR creation for parent component", func() {
```

Comment:
I guess this is a consequence of the parent component has PaC provision via the annotation?

Reply:
Yes